From 5392e2573abea13a461400308f0fa4b5af097139 Mon Sep 17 00:00:00 2001
From: David Daney <david.daney@cavium.com>
Date: Tue, 16 Oct 2012 11:19:54 -0700
Subject: [PATCH 023/122] MIPS: Align stack top for
 CONFIG_MIPS_HUGE_TLB_SUPPORT.

Source: Cavium Networks, Inc.
MR: 00000
Type: Integration
Disposition: Merged from Octeon Tree
ChangeID: ebf87bed4051a5b54e6ee0e38b87e794219afa91
Description:

The alignment is beneficial for both HUGETLB_PAGE and
TRANSPARENT_HUGEPAGE.

Signed-off-by: David Daney <david.daney@cavium.com>
Signed-off-by: Leonid Rosenboim <lrosenboim@caviumnetworks.com>
Signed-off-by: Corey Minyard <cminyard@mvista.com>
[Original patch taken from Cavium SDK 3.1.1 525]
Signed-off-by: Bin Jiang <bin.jiang@windriver.com>
---
 arch/mips/include/asm/processor.h | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/arch/mips/include/asm/processor.h b/arch/mips/include/asm/processor.h
index f4fa4a8..4ab5e2c 100644
--- a/arch/mips/include/asm/processor.h
+++ b/arch/mips/include/asm/processor.h
@@ -84,7 +84,7 @@ extern unsigned int vced_count, vcei_count;
 
 #define TASK_IS_32BIT_ADDR test_thread_flag(TIF_32BIT_ADDR)
 
-# ifdef CONFIG_HUGETLB_PAGE
+# ifdef CONFIG_MIPS_HUGE_TLB_SUPPORT
 /*
  * Align the STACK_TOP on a HPAGE_SIZE boundry so the stack may be
  * remapped to a huge page.
@@ -92,7 +92,7 @@ extern unsigned int vced_count, vcei_count;
 # define SPECIAL_PAGES_BASE ((TASK_SIZE & PAGE_MASK) - SPECIAL_PAGES_SIZE)
 # define STACK_TOP (SPECIAL_PAGES_BASE & HPAGE_MASK)
 
-# endif /* CONFIG_HUGETLB_PAGE */
+# endif /* CONFIG_MIPS_HUGE_TLB_SUPPORT */
 #endif /* CONFIG_64BIT */
 
 #ifndef STACK_TOP
-- 
1.8.2.1

